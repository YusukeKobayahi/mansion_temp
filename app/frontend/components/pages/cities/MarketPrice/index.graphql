fragment MsLibraryCitiesMarketPrice on City {
  name
  prefecture {
    name
    pricePredictions {
      salePricePerTsubo {
        date
        prefecture
      }
      salePricePerSquare {
        date
        prefecture
      }
    }
  }
  pricePredictions {
    salePricePerTsubo {
      date
      city
    }
    salePricePerSquare {
      date
      city
    }
  }
}
