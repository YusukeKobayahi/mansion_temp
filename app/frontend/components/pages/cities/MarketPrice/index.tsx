import BoxHeading from "~/components/commons/headings/BoxHeading";
import { MsLibraryCitiesMarketPriceFragment } from "~/graphql/generated";
import { createKey } from "~/lib/utils";
import dynamic from "next/dynamic";
const MarketPriceItem = dynamic(
  () => import("~/components/commons/Chart/MarketPriceItem")
);

type MarketPriceProps = {
  data: MsLibraryCitiesMarketPriceFragment;
};

const MarketPrice: React.FC<MarketPriceProps> = ({
  data,
}: MarketPriceProps) => {
  const defaultDiisplayKeyList = ["city", "prefecture"];
  const citySalePricePerSquare = data.pricePredictions.salePricePerSquare;
  const citySalePricePerTsubo = data.pricePredictions.salePricePerTsubo;
  const prefectureSalePricePerSquare =
    data.prefecture.pricePredictions.salePricePerSquare;
  const prefectureSalePricePerTsubo =
    data.prefecture.pricePredictions.salePricePerTsubo;

  if (
    ![
      citySalePricePerSquare,
      citySalePricePerTsubo,
      prefectureSalePricePerSquare,
      prefectureSalePricePerTsubo,
    ].every((el) => el && el.length !== 0)
  )
    return null;

  const salePricePerSquare = [];
  for (let i = 0; i < citySalePricePerSquare.length; i++) {
    salePricePerSquare.push({
      date: citySalePricePerSquare[i].date,
      city: Number(citySalePricePerSquare[i].city),
      prefecture: Number(prefectureSalePricePerSquare[i].prefecture),
    });
  }
  const salePricePerTsubo = [];
  for (let i = 0; i < citySalePricePerTsubo.length; i++) {
    salePricePerTsubo.push({
      date: citySalePricePerTsubo[i].date,
      city: Number(citySalePricePerTsubo[i].city),
      prefecture: Number(prefectureSalePricePerTsubo[i].prefecture),
    });
  }

  const prefectureName = data.prefecture.name;
  const cityName = data.name;

  const name = prefectureName + cityName;

  const targetListItems = [
    {
      name: prefectureName + cityName,
      key: createKey("city"),
      color: "#686D87",
    },
    {
      name: prefectureName,
      key: createKey("prefecture"),
      color: "#505050",
    },
  ];

  const mainKey = "city";

  return (
    <section className={"m_y_30px"}>
      <h2>
        <BoxHeading>{name}周辺のマンション相場遷移</BoxHeading>
      </h2>
      <MarketPriceItem
        mainKey={mainKey}
        name={name}
        salePricePerSquare={salePricePerSquare}
        salePricePerTsubo={salePricePerTsubo}
        targetListItems={targetListItems}
        defaultDiisplayKeyList={defaultDiisplayKeyList}
      />
    </section>
  );
};

export default MarketPrice;
