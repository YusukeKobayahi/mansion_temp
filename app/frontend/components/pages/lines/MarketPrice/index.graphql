fragment MsLibraryLinesMarketPrice on RailwayLine {
  name
  pricePredictions {
    salePricePerTsubo {
      date
      line
    }
    salePricePerSquare {
      date
      line
    }
  }
}
