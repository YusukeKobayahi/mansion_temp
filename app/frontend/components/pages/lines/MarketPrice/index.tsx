import BoxHeading from "~/components/commons/headings/BoxHeading";
import { MsLibraryLinesMarketPriceFragment } from "~/graphql/generated";
import { createKey } from "~/lib/utils";
import dynamic from "next/dynamic";
const MarketPriceItem = dynamic(
  () => import("~/components/commons/Chart/MarketPriceItem")
);

type MarketPriceProps = {
  data: MsLibraryLinesMarketPriceFragment;
};

const MarketPrice: React.FC<MarketPriceProps> = ({
  data,
}: MarketPriceProps) => {
  const defaultDiisplayKeyList = ["line"];
  const lineSalePricePerSquare = data.pricePredictions.salePricePerSquare;
  const lineSalePricePerTsubo = data.pricePredictions.salePricePerTsubo;

  if (
    ![lineSalePricePerSquare, lineSalePricePerTsubo].every(
      (el) => el && el.length !== 0
    )
  )
    return null;

  const salePricePerSquare = [];
  for (let i = 0; i < lineSalePricePerSquare.length; i++) {
    salePricePerSquare.push({
      date: lineSalePricePerSquare[i].date,
      line: Number(lineSalePricePerSquare[i].line),
    });
  }
  const salePricePerTsubo = [];
  for (let i = 0; i < lineSalePricePerTsubo.length; i++) {
    salePricePerTsubo.push({
      date: lineSalePricePerTsubo[i].date,
      line: Number(lineSalePricePerTsubo[i].line),
    });
  }

  const lineName = data.name;

  const targetListItems = [
    {
      name: lineName,
      key: createKey("line"),
      color: "#686D87",
    },
  ];

  const mainKey = "line";

  return (
    <section className={"m_y_30px"}>
      <h2>
        <BoxHeading>{lineName}周辺のマンション相場遷移</BoxHeading>
      </h2>
      <MarketPriceItem
        mainKey={mainKey}
        name={lineName}
        salePricePerSquare={salePricePerSquare}
        salePricePerTsubo={salePricePerTsubo}
        targetListItems={targetListItems}
        defaultDiisplayKeyList={defaultDiisplayKeyList}
      />
    </section>
  );
};

export default MarketPrice;
