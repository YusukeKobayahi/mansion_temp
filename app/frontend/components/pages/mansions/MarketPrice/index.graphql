fragment MsLibraryMansionsMarketPrice on Mansion {
  name
  prefecture {
    name
  }
  city {
    name
  }
  town {
    name
  }
  access {
    stationName
  }
  priceHubble {
    salePricePerSquare {
      date
      building
      station
      prefecture
      city
      town
    }
    salePricePerTsubo {
      date
      building
      station
      prefecture
      city
      town
    }
  }
}
