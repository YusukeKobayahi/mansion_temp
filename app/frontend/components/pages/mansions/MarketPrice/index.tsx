import { MsLibraryMansionsMarketPriceFragment } from "~/graphql/generated";
import dynamic from "next/dynamic";
const MarketPriceItem = dynamic(
  () => import("~/components/commons/Chart/MarketPriceItem")
);

type MarketPriceProps = {
  data: MsLibraryMansionsMarketPriceFragment;
};

const MarketPrice: React.FC<MarketPriceProps> = ({
  data,
}: MarketPriceProps) => {
  const defaultDiisplayKeyList = [
    "building",
    "town",
    "city",
    "prefecture",
    "station",
  ];
  const salePricePerSquare = data.priceHubble?.salePricePerSquare.map((o) => {
    return {
      date: o.date,
      building: Number(o.building),
      town: Number(o.town),
      city: Number(o.city),
      prefecture: Number(o.prefecture),
      station: Number(o.station),
    };
  });
  const salePricePerTsubo = data.priceHubble?.salePricePerTsubo.map((o) => {
    return {
      date: o.date,
      building: Number(o.building),
      town: Number(o.town),
      city: Number(o.city),
      prefecture: Number(o.prefecture),
      station: Number(o.station),
    };
  });
  if (
    !salePricePerSquare ||
    salePricePerSquare.length === 0 ||
    !salePricePerTsubo ||
    salePricePerTsubo.length === 0
  )
    return null;

  const prefectureName = data.prefecture.name;
  const cityName = data.city.name;
  const townName = data.town.name;
  const stationName =
    data.access.length > 0
      ? data.access.map((a) => a.stationName + "é§…")[0]
      : "";

  const name = prefectureName + cityName + townName;

  const createKey = (
    s: "building" | "town" | "city" | "prefecture" | "station" | "line"
  ): "building" | "town" | "city" | "prefecture" | "station" | "line" => s;

  const targetListItems = [
    {
      name: data.name,
      key: createKey("building"),
      color: "#B2A173",
    },
    {
      name: prefectureName + cityName + townName,
      key: createKey("town"),
      color: "#746744",
    },
    {
      name: prefectureName + cityName,
      key: createKey("city"),
      color: "#686D87",
    },
    {
      name: prefectureName,
      key: createKey("prefecture"),
      color: "#505050",
    },
    {
      name: stationName,
      key: createKey("station"),
      color: "#9D9990",
    },
  ];

  const mainKey = "town";

  return (
    <MarketPriceItem
      mainKey={mainKey}
      name={name}
      salePricePerSquare={salePricePerSquare}
      salePricePerTsubo={salePricePerTsubo}
      targetListItems={targetListItems}
      defaultDiisplayKeyList={defaultDiisplayKeyList}
    />
  );
};

export default MarketPrice;
