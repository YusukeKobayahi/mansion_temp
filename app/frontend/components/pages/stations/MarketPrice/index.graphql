fragment MsLibraryStationsMarketPrice on RailwayStation {
  name
  pricePredictions {
    salePricePerTsubo {
      date
      station
    }
    salePricePerSquare {
      date
      station
    }
  }
  city {
    name
    pricePredictions {
      salePricePerTsubo {
        date
        city
      }
      salePricePerSquare {
        date
        city
      }
    }
    prefecture {
      name
      pricePredictions {
        salePricePerTsubo {
          date
          prefecture
        }
        salePricePerSquare {
          date
          prefecture
        }
      }
    }
  }
}
