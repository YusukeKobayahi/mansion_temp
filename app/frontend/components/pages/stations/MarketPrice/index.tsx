import BoxHeading from "~/components/commons/headings/BoxHeading";
import { MsLibraryStationsMarketPriceFragment } from "~/graphql/generated";
import { createKey } from "~/lib/utils";
import dynamic from "next/dynamic";
const MarketPriceItem = dynamic(
  () => import("~/components/commons/Chart/MarketPriceItem")
);

type MarketPriceProps = {
  data: MsLibraryStationsMarketPriceFragment;
};

const MarketPrice: React.FC<MarketPriceProps> = ({
  data,
}: MarketPriceProps) => {
  const defaultDiisplayKeyList = ["station", "city", "prefecture"];
  const stationSalePricePerSquare = data.pricePredictions.salePricePerSquare;
  const stationSalePricePerTsubo = data.pricePredictions.salePricePerTsubo;
  const citySalePricePerSquare = data.city.pricePredictions.salePricePerSquare;
  const citySalePricePerTsubo = data.city.pricePredictions.salePricePerTsubo;
  const prefectureSalePricePerSquare =
    data.city.prefecture.pricePredictions.salePricePerSquare;
  const prefectureSalePricePerTsubo =
    data.city.prefecture.pricePredictions.salePricePerTsubo;

  if (
    ![
      stationSalePricePerSquare,
      stationSalePricePerTsubo,
      citySalePricePerSquare,
      citySalePricePerTsubo,
      prefectureSalePricePerSquare,
      prefectureSalePricePerTsubo,
    ].every((el) => el && el.length !== 0)
  )
    return null;

  const salePricePerSquare = [];
  for (let i = 0; i < stationSalePricePerSquare.length; i++) {
    salePricePerSquare.push({
      date: stationSalePricePerSquare[i].date,
      station: Number(stationSalePricePerSquare[i].station),
      city: Number(citySalePricePerSquare[i].city),
      prefecture: Number(prefectureSalePricePerSquare[i].prefecture),
    });
  }
  const salePricePerTsubo = [];
  for (let i = 0; i < stationSalePricePerTsubo.length; i++) {
    salePricePerTsubo.push({
      date: stationSalePricePerTsubo[i].date,
      station: Number(stationSalePricePerTsubo[i].station),
      city: Number(citySalePricePerTsubo[i].city),
      prefecture: Number(prefectureSalePricePerTsubo[i].prefecture),
    });
  }

  const stationName = data.name + "駅";
  const cityName = data.city.name;
  const prefectureName = data.city.prefecture.name;

  const targetListItems = [
    {
      name: stationName,
      key: createKey("station"),
      color: "#9D9990",
    },
    {
      name: cityName,
      key: createKey("city"),
      color: "#686D87",
    },
    {
      name: prefectureName,
      key: createKey("prefecture"),
      color: "#505050",
    },
  ];

  const mainKey = "station";

  return (
    <section className={"m_y_30px"}>
      <h2>
        <BoxHeading>{stationName}周辺のマンション相場遷移</BoxHeading>
      </h2>
      <MarketPriceItem
        mainKey={mainKey}
        name={stationName}
        salePricePerSquare={salePricePerSquare}
        salePricePerTsubo={salePricePerTsubo}
        targetListItems={targetListItems}
        defaultDiisplayKeyList={defaultDiisplayKeyList}
      />
    </section>
  );
};

export default MarketPrice;
