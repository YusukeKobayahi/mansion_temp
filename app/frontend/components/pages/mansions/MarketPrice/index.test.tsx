import { render } from "@testing-library/react";
import mockData from "~/components/pages/mansions/fixtures/mockData.json";
import { MansionQuery } from "~/graphql/generated";
import MarketPrice from "~/components/pages/mansions/MarketPrice";

describe("components/pages/mansions/MarketPrice", () => {
  it("正常に表示される", () => {
    const mock = mockData as MansionQuery;
    const { asFragment } = render(<MarketPrice data={mock.mansion} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
