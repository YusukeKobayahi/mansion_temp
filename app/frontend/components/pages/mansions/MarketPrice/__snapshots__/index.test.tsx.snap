// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/mansions/MarketPrice 正常に表示される 1`] = `<DocumentFragment />`;
